stock_outgoing_shipment_report/report/stock_outgoing_shipment_report.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2020 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+FIELDS_PROPERTIES = {
+    "shipping_ref_code": ["Char", 2],
+    "carrier_name": ["Char", 20],
+    "partner_ref": ["Char", 10],
+    "partner_name": ["Char", 32],
+    "partner_zip": ["Char", 8],
+    "partner_address": ["Char", 80],
+    "product_code": ["Char", 7],
+    "product_name": ["Char", 32],
+    "case_qty": ["Float", 5],
+    "separate_qty": ["Float", 5],
+    "lot_num": ["Float", 6],
+    "lot_branch_num": ["Float", 2],
+    "delivery_division": ["Char", 1],
+    "customer_delivery_note": ["Char", 9],
+    "client_order_ref": ["Char", 30],
+    "memo": ["Char", 9],
+}
+
+
+class StockOutgoingShipmentReport(models.TransientModel):
+    _name = "stock.outgoing.shipment.report"
+
+    move_id = fields.Many2one("stock.move", string="Stock Move", readonly=True,)
+    delivery_date = fields.Char(
+        string="Delivery Date", compute="_compute_date_fields", store=True
+    )
+    expected_date = fields.Char(
+        string="Expected Date", compute="_compute_date_fields", store=True
+    )
+    shipping_ref_code = fields.Char("Shipping Ref Code")
+    carrier_name = fields.Char("Carrier")
+    partner_name = fields.Char("Customer")
+    partner_ref = fields.Char("Customer Code")
+    partner_zip = fields.Char("Zip")
+    partner_address = fields.Char("Address")
+    partner_phone = fields.Char("Phone")
+    product_code = fields.Char("Product Code")
+    product_name = fields.Char("Product Name")
+    case_qty = fields.Float("Case Quantity")
+    separate_qty = fields.Char("Separate Quantity")
+    expiry_date_edit = fields.Date("Expiry Date (Edit)")
+    expiry_date = fields.Char(
+        string="Expiry Date", compute="_compute_date_fields", store=True
+    )
+    lot_num = fields.Char("Warehouse Company Lot")
+    lot_branch_num = fields.Char("Warehouse Company Lot Branch")
+    delivery_division = fields.Char("Delivery Division")
+    customer_delivery_note = fields.Char("Customer Delivery Note")
+    client_order_ref = fields.Char("Customer Reference")
+    memo = fields.Char("Memo")
+
+    @api.multi
+    @api.depends("move_id.date_delivered", "move_id.date_expected", "expiry_date_edit")
+    def _compute_date_fields(self):
+        for line in self:
+            date_format = "%Y/%m/%d"
+            if line.move_id.date_delivered:
+                line.delivery_date = line.move_id.date_delivered.strftime(date_format)
+            if line.move_id.date_expected:
+                line.expected_date = line.move_id.date_expected.strftime(date_format)
```

Comment:
For delivery_date, use move_id.picking_id.delivery_due_date, and convert the datetime to user's timezone first.